collect/collect.go

```diff
@@ -1620,3 +1612,72 @@ func (i *InMemCollector) publishTraceDecision(ctx context.Context, td TraceDecis
 		}).Logf("Failed to publish trace decision")
 	}
 }
+
+func (i *InMemCollector) sendDropDecisions() {
+	if i.Config.GetCollectionConfig().EnableTraceLocality {
+		return
+	}
+
+	tickerInterval := time.Duration(i.Config.GetCollectionConfig().DropDecisionSendInterval)
+	if i.Config.GetCollectionConfig().DropDecisionSendInterval == 0 {
+		tickerInterval = defaultDropDecisionTicker
+	}
+
+	ticker := i.Clock.NewTicker(tickerInterval)
+	defer ticker.Stop()
+	traceIDs := make([]string, 0, i.Config.GetCollectionConfig().MaxDropDecisionBatchSize)
+	send := false
+	eg := &errgroup.Group{}
+	for {
+		select {
+		case <-i.done:
+			eg.Wait()
+			return
+		case id, ok := <-i.dropDecisionBatch:
+			if !ok {
+				eg.Wait()
+				return
+			}
+			// if we get a trace ID, add it to the list
+			traceIDs = append(traceIDs, id)
+			// if we exceeded the max count, we need to send
+			if len(traceIDs) >= i.Config.GetCollectionConfig().MaxDropDecisionBatchSize {
+				send = true
+			}
+		case <-ticker.Chan():
```

Comment:
This gives us a possible situation like this:
* Max size is reached at time T
* We send the batch
* We get one decision
* At T+1ms the ticker fires
* We send that batch

If the number of decisions per second is close to the max batch size, then the above is very likely to happen.

I've thought of two possible ways to deal with this:
* Keep a timestamp of the oldest item in the batch; if it's less than our ticker time, then don't send yet.
* Don't use a ticker, use a timer, and restart the timer after every send, no matter why we sent it.

What do you think?

Reply:
I like the timer idea. I will update the code to do that